text/0758-default-component-manager.md

```diff
@@ -0,0 +1,138 @@
+---
+Stage: Accepted
+Start Date: 2021-05-17
+Release Date: Unreleased
+Release Versions:
+  ember-source: vX.Y.Z
+  ember-data: vX.Y.Z
+Relevant Team(s): Ember.js, Learning
+RFC PR: https://github.com/emberjs/rfcs/pull/758
+---
+
+# Default Component Manager
+
+## Summary
+
+Anything that can be in a template has its lifecycle managed by the manager pattern.
+Today, we have known managers for `@glimmer/component`, `@ember/helper`, etc.
+But what happens when the VM encounters an object for which there is no manager?,
+such as a plain function? This RFC explores and proposes a default behavior for
+those unknown scenarios when it comes to _components_.
+
+## Motivation
+
+The addon, [ember-could-get-used-to-this](https://github.com/pzuraq/ember-could-get-used-to-this)
+demonstrated that it's possible to use plain functions for helpers and modifiers.
+Components on the other hand, have had no such "plain" / "vanilla" support / adoption from addons.
+Maybe it's worth exploring what default behavior components should have.
+
+This could reduce the number of dependencies folks need to worry about when building components.
```

Comment:
If this is aimed at reducing the number of imports, then it has an interesting side effect: it makes the “base case” *more* for people to type. It's extremely rare in my experience to want a backing class which *doesn't* care at all about `args`. Given that, the proposed delta for your average component would be from today’s default—

```js
import Component from '@glimmer/component';
import { tracked } from '@glimmer/tracking';

export default class MyComponent extends Component {
  // whatever local and derived state
}
```

—to this:

```js
import { tracked } from '@glimmer/tracking';

export default class MyComponent {
  constructor(owner, args) {
    this.owner = owner;
    this.args = args;
  }

  // whatever local and derived state
}
```

This is a substantial *increase* in lines of code, and the result is a substantial *decrease* in the basic safety: you can accidentally end up just stomping `this.args` somewhere by writing `this.args = 'lol';`: something Glimmer component protects you from.

Reply:
that's a good point :thinking: